src/adapters/philips_hue/hub.rs

```diff
@@ -70,13 +69,28 @@ impl<C: Controller> Hub<C> {
                         adapter.controller.adapter_notification(
                             json_value!({ adapter: "philips_hue",
                                 message: "NeedsPairing", hub: id }));
-                        if api.try_pairing() {
-                            break;
+                        let pairing_result = api.lock().unwrap().try_pairing();
+                        match pairing_result {
+                            Ok(Some(new_token)) => {
+                                info!("Pairing success with Philips Hue Bridge {}", id);
+                                // Save the new token
+                                adapter.controller.get_config().set(
+                                    "philips_hue",
+                                    &format!("token_{}", id),
+                                    &new_token);
+                                api.lock().unwrap().update_token(&new_token);
+                                break;
+                            },
+                            Ok(None) => {
+                                info!("Push pairing button on Philips Hue Bridge {}", id);
```

Comment:
We first display a warn!("Philips Hue detected but not paired...") and here an info!(). warn!() are a bit easier to spot as a call to action I think.